pynamodb/attributes.py

```diff
@@ -376,65 +348,42 @@ class BinarySetAttribute(SetMixin, Attribute[Set[bytes]]):
 
     def serialize(self, value):
         """
-        Returns a base64 encoded binary string
+        Returns a list of base64 encoded binary strings. Encodes empty sets as "None".
         """
-        if value and len(value):
-            return [b64encode(val).decode(DEFAULT_ENCODING) for val in sorted(value)]
-        else:
-            return None
+        return [b64encode(v).decode(DEFAULT_ENCODING) for v in value] if value else None
 
     def deserialize(self, value):
         """
-        Returns a decoded string from base64
+        Returns a set of decoded byte strings from base64 encoded values.
         """
-        try:
-            if value and len(value):
-                return {b64decode(val.decode(DEFAULT_ENCODING)) for val in value}
-        except AttributeError:
-            return {b64decode(val) for val in value}
+        return {b64decode(v) for v in value}
```

Comment:
In python 3 b64decode takes bytes and string objects so this removes the `decode`.
Removes the check if `value` is `None` to enforce decoding only DynamoDB supported values for the `BS` type.